app/models/order_detail.rb

```diff
@@ -396,6 +397,10 @@ def self.ordered_or_reserved_in_range(start_date, end_date)
     end
   end
 
+  def is_canceled? # non-standard name to avoid conflict with AASM
+    canceled_at.present?
```

Comment:
I was wondering what the purpose was of having `canceled` in the state machine and also separate `canceled_at` fields. Makes the logic fairly convoluted just to keep the data consistent.